### ISSUE TITLE
ts-loader should ignore require() statements in definition (.d.ts) files

### ISSUE THREAD

**Issue by stkb** [jbrantly/ts-loader#4]
Some definition files, for example `node.d.ts` from DefinitelyTyped, contain named external modules, eg
``` typescript
declare module "events" { ... }
```
and then other named external modules that reference these modules with a ```require``` statement, eg
``` typescript
declare module "http" {
    import events = require("events");
    ...
}
```

Ts-loader then tries to resolve these modules through webpack, which it shouldn't be doing, because this is still all just metadata and they're not yet being used. It should only try to resolve them once a such a module is being `require`'d in a code (*.ts) file.

It causes the build to fail if those modules aren't available (which is the case for some of the node modules like 'events', 'net' etc.)

This makes it impossible to use node.d.ts in a project with ts-loader, but this problem isn't limited to that file; it can happen with any definition file that follows the same pattern.

Reproduction is easy with an `app.ts` file that just references a `node.d.ts` file:
``` typescript
/// <reference path="node.d.ts" />
```
and then a `webpack.config.js` file:
``` javascript
module.exports = {
    entry: './app.ts',
    output: {
        path: __dirname,
        filename: 'bundle.js',
    },
    resolve: {
        extensions: ['', '.webpack.js', '.web.js', '.js', '.ts']
    },
    module: {
        loaders: [
            { test: /\.ts$/, loader: 'ts-loader?sourceMap&target=ES5' }
        ]
    },
}
```

**Comment by jbrantly** [comment]
Thanks for the detailed report! I'll take a closer look and get a fix out hopefully later today.

**Comment by jbrantly** [comment]
I've taken a look at this and I think I agree that we shouldn't resolve modules from import statements in declaration files. According to the spec only top-level modules can be imported at this point anyways[1], so it's pretty much always going to fail. I've committed if you'd like to try it. I'll publish to npm after I resolve the other issue you reported.

[1] TS Spec section 12.2
> An ExternalImportDeclaration in an AmbientExternalModuleDeclaration may reference other external
modules only through top-level external module names. Relative external module names are not
permitted.

**Comment by stkb** [comment]
Thanks! From quick testing it seems to be fixed :)

**Comment by jbrantly** [comment]
This is now fixed with v0.3.0
